### PR TITLE
libgit-rs: add get_bool(), get_ulong(), and get_pathname() methods to ConfigSet

### PR DESCRIPTION
**Pull request by ionnss**
## libgit-rs: Adding new wrappers to ConfigSet API

This series expands the Rust ConfigSet API with support for additional configuration value types, following Git's established patterns by wrapping native C functions.

### Implementation

Adds three new ConfigSet methods to the Rust bindings:
- `get_bool()`: Parse boolean values using `git_configset_get_bool()`
- `get_ulong()`: Parse unsigned long integers using `git_configset_get_ulong()`  
- `get_pathname()`: Parse file paths using `git_configset_get_pathname()`, returning PathBuf

All methods follow the existing pattern of `get_int()` and `get_string()`, ensuring consistent behavior with Git's native parsing logic.

### Changes
- Fixed missing wrapper functions in public_symbol_export.[ch] (Philip's feedback)
- Rebased to correct base commit bb69721404 (Junio's feedback)
- Squashed into single clean commit as requested
- Removed unrelated README.md commit
- Fixed commit message formatting per GitGitGadget requirements

The Rust ConfigSet API now supports 5 configuration value types: int, string, bool, ulong, and pathname.

cc: Phillip Wood <phillip.wood123@gmail.com>
cc: "brian m. carlson" <sandals@crustytoothpaste.net>

### CODE REVIEW

File: contrib/libgit-rs/src/config.rs [hunk 1]
```diff
@@ -68,6 +68,26 @@ impl ConfigSet {
             Some(owned_str)
```
[the remark this answers is not in the document]
Comment (ionnss) [reply contrib/libgit-rs/src/config.rs:72]: Got it. Will work on that!

**Comment by ionnss** [comment]
Updated to address review feedback:

- Now uses `git_configset_get_bool()` C function instead of reimplementing boolean parsing in Rust
- Fixed function signature in libgit-sys bindings  
- Improved error messages
- Added comprehensive tests for edge cases like "00" and "100"

This ensures the Rust API always matches Git's actual boolean parsing behavior.

Thank you for the feedback and patience.

**Comment by ionnss** [comment]
Added two more ConfigSet methods based on the same architectural pattern:

**get_ulong()**: Handles large unsigned integers (> 32-bit) using `git_configset_get_ulong()`
- Useful for config values like pack size limits, timeouts, etc.
- Test case includes 4294967296 (4GB) which exceeds 32-bit int range

**get_pathname()**: Handles file paths using `git_configset_get_pathname()`  
- Returns PathBuf for type-safe path handling
- Useful for config values like core.editor, core.excludesfile, etc.

Both functions follow the established pattern of wrapping Git's C functions rather than reimplementing parsing logic, ensuring consistency with Git's behavior.

The ConfigSet API now supports 5 different value types: int, string, bool, ulong, and pathname.

**Comment by ionnss** [comment]
Rebased and cleaned up commit history as requested:

**Commit 1**: `libgit-rs: add get_bool() method to ConfigSet`
- Uses git_configset_get_bool() C function for consistent parsing
- Handles all Git boolean formats and edge cases correctly
- Includes comprehensive tests

**Commit 2**: `libgit-rs: add get_ulong() and get_pathname() methods`  
- Extends ConfigSet API with additional value types
- Follows same architectural pattern as get_bool()

Separated the unrelated README.md fix as suggested. The series now presents clean, professional commits focused on the final solutions.

**Comment by dscho** [comment]
> Added two more ConfigSet methods based on the same architectural pattern:
> 
> **get_ulong()**: Handles large unsigned integers (> 32-bit) using `git_configset_get_ulong()`
> 
> * Useful for config values like pack size limits, timeouts, etc.
> * Test case includes 4294967296 (4GB) which exceeds 32-bit int range
> 
> **get_pathname()**: Handles file paths using `git_configset_get_pathname()`
> 
> * Returns PathBuf for type-safe path handling
> * Useful for config values like core.editor, core.excludesfile, etc.
> 
> Both functions follow the established pattern of wrapping Git's C functions rather than reimplementing parsing logic, ensuring consistency with Git's behavior.
> 
> The ConfigSet API now supports 5 different value types: int, string, bool, ulong, and pathname.

@ionnss I am afraid that you will need to reply on the mailing list as described here: https://gitgitgadget.github.io/reply-to-this

If you want to describe changes between iterations instead, please edit [the PR description](https://github.com/gitgitgadget/git/pull/1977#issue-3451455481) (which will be sent as cover letter).

**Comment by ionnss** [comment]
/submit

## Changes in v4:
- Fixed missing wrapper functions in public_symbol_export.[ch] (addresses Philip's feedback)
- Rebased to correct base commit bb69721404 (addresses Junio's feedback)  
- Squashed into single clean commit as requested
- Removed unrelated README.md commit
- Fixed commit message formatting per GitGitGadget requirements

This version includes all three ConfigSet methods (get_bool, get_ulong, get_pathname) with proper FFI wrapper functions and comprehensive tests.

**Comment by dscho** [comment]
> /submit
> 
> ## Changes in v4:
> * Fixed missing wrapper functions in public_symbol_export.[ch] (addresses Philip's feedback)
> * Rebased to correct base commit [bb69721](https://github.com/gitgitgadget/git/commit/bb69721404348ea2db0a081c41ab6ebfe75bdec8) (addresses Junio's feedback)
> * Squashed into single clean commit as requested
> * Removed unrelated README.md commit
> * Fixed commit message formatting per GitGitGadget requirements
> 
> This version includes all three ConfigSet methods (get_bool, get_ulong, get_pathname) with proper FFI wrapper functions and comprehensive tests.

@ionnss that's not how it works.

You need to edit the PR description, i.e. the initial comment of the PR, to add that `## Changes` section. And then you issue a single `/submit` command, without any arguments.
